Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/file_bugs.py

```diff
@@ -0,0 +1,197 @@
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import os
+import re
+import sys
+
+from .command import Command
+from .pull_request import PullRequest
+
+from webkitbugspy import Tracker, radar
+from webkitcorepy import arguments, run, Terminal
+from webkitscmpy import local, log, remote
+from webkitpy.common.system.user import User
+
+from webkitbugspy.radar import Tracker as RadarTracker
+from webkitbugspy.bugzilla import Tracker as BugzillaTracker
+
+
+class FileBugs(Command):
+    name = 'file-bugs'
+    aliases = ['bugs']
+    help = 'File bugzilla issues for commits without one specified'
+
+    BUGZILLA_PLACEHOLDER = 'Need the bug URL (OOPS!).'
+    RADAR_PLACEHOLDER = 'Include a Radar link (OOPS!).'
+
+    @classmethod
+    def parser(cls, parser, loggers=None):
+        parser.add_argument(
+            '--remote', dest='remote', type=str, default=None,
+            help='Make a pull request against a specific remote',
+        )
+
+        # Probably want some more args to configure the behaviour around
+        # auto-editing commit messages etc
+
+    @classmethod
+    def parse_description(cls, lines):
+        # If there's less than 6 lines it can't really match the commit template
+        if len(lines) < 6:
+            return
+
+        # Check if there's something on the 3rd line (for a rdar:)
+        # If so, then start our search for the description one line
+        # later.
+        desc_start = 5
+        if not lines[2] == '':
+            desc_start = 6
+            if len(lines) < 7:
+                return
+
+        # Search for a description from the 7th line until we see a '*' for file changes.
+        # (should be title,bugzilla,rdar?,blank,reviewer,blank on first 5/6 lines).
+        # FIXME: This is very hardcoded to the exact template, could probably be a bit more
+        # adaptive.
+        desc = []
+        for line in lines[desc_start:]:
+            if line.startswith('* '):
+                break
+            desc.append(line)
+        return "\n".join(desc)
+
+    @classmethod
+    def create_bug(cls, repository, commit):
+        lines = commit.message.splitlines()
+        if len(lines) < 2:
+            return
+
+        issue = None
+        radar = None
+        needs_radar = False
+        if cls.BUGZILLA_PLACEHOLDER not in lines[1]:
```

Comment:
I think for this bit, you want the project's default issue tracker (can get that from `Tracker.instance()`)